internal/website/content/howto/docstore/actions.md

```diff
@@ -0,0 +1,104 @@
+---
+title: "Read and Write Documents"
+date: 2019-06-11T10:11:59-04:00
+draft: true
+weight: 2
+---
+
+In Docstore, you use actions to read, modify and write documents. You can
+execute a single action, or run multiple actions together in an _action list_.
+
+<!--more-->
+
+## Representing Documents
+
+We'll use a collection with documents represented by this Go struct:
+
+```go
+type Player struct {
+	Name             string
+	Score            int
+	DocstoreRevision interface{}
+}
+```
+
+We recommend using structs for documents because they impose some structure on
+your data, but Docstore also accepts `map[string]interface{}` values. See
+[the `docstore` package
+documentation](https://godoc.org/gocloud.dev/internal/docstore#hdr-Documents)
+for more information.
+
+The `DocstoreRevision` field holds information about the latest revision of the
+document. We discuss it [below]({{< ref "#docrev" >}}).
+
+## Actions and Action Lists
+
+Once you have [opened a collection]({{< ref "open-collection.md" >}}), you
+can call action methods on it. We will use `coll` as the variable holding the collection.
+
+Docstore supports six actions on documents:
+
+- `Get` retrieves a document.
+- `Create` creates a new document.
+- `Replace` replaces an existing document.
+- `Put` puts a document whether or not it is already present.
+- `Update` applies a set of modifications to a document.
+- `Delete` deletes a document.
+
+You can create a single document with the `Collection.Create` method:
+
+```go
+err := coll.Create(ctx, &Player{Name: "Pat", Score: 10})
+if err != nil {
+    return err
+}
+```
+
+If you have more than one action to perform, it is better to use an action
+list. Drivers can optimize action lists by using bulk RPCs, running the actions
+concurrently, or employing a provider's special features to improve efficiency
+and reduce cost. Here we create several documents using an action list.
+
+{{< goexample "gocloud.dev/internal/docstore.ExampleCollection_Actions_bulkWrite" >}}
+
+`ActionList` has a fluent API, so you can build and execute a sequence of
+actions in one line of code. Here we `Put` a document and immediately `Get` its new
```

Comment:
Should we mention this example may not get the new value depending on whether the service supports it?

Reply:
Done.